### PR TITLE
Issue 46

### PR DESCRIPTION
**Pull request by cat-bug**

### CODE REVIEW

File: src/main/java/org/testmonkeys/jentitytest/comparison/property/StringComparator.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import org.testmonkeys.jentitytest.comparison.ComparisonContext;
+import org.testmonkeys.jentitytest.comparison.SingleResultComparator;
+import org.testmonkeys.jentitytest.comparison.conditionalChecks.NullConditionalCheck;
+import org.testmonkeys.jentitytest.comparison.result.ComparisonResult;
+import org.testmonkeys.jentitytest.framework.JEntityTestException;
+
+public class StringComparator extends SingleResultComparator {
+    private boolean caseSensitive = true;
+    private boolean ignoreTrailing = false;
```
Comment (costea32) [inline src/main/java/org/testmonkeys/jentitytest/comparison/property/StringComparator.java]: I think a better name would be <trim>

File: src/main/java/org/testmonkeys/jentitytest/comparison/property/StringComparator.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            expected = (String) expectedValue;
+        } catch (ClassCastException castException) {
+            throw new JEntityTestException("Expected and Actual values must of type " + String.class.getName(),
+                    castException);
+        }
+
+        if (actual != null && expected != null) {
+            for (char ignore : ignoreCharacters) {
+
+                String ignoreString = String.valueOf(ignore);
```
Comment (costea32) [inline src/main/java/org/testmonkeys/jentitytest/comparison/property/StringComparator.java]: we should be using the value of ignore directly in replace, no need to transform to String

File: src/main/java/org/testmonkeys/jentitytest/comparison/property/StringComparator.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                actual = actual.replace(ignoreString, "");
+                expected = expected.replaceAll(ignoreString, "");
```
Comment (costea32) [inline src/main/java/org/testmonkeys/jentitytest/comparison/property/StringComparator.java]: change to replace()

File: src/main/java/org/testmonkeys/jentitytest/comparison/property/StringComparator.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+        try {
+            actual = (String) actualValue;
+            expected = (String) expectedValue;
+        } catch (ClassCastException castException) {
+            throw new JEntityTestException("Expected and Actual values must of type " + String.class.getName(),
+                    castException);
+        }
+
+        if (actual != null && expected != null) {
```
Comment (cat-bug) [inline src/main/java/org/testmonkeys/jentitytest/comparison/property/StringComparator.java]: related to #50

**Comment by costea32** [comment]
closing issue #49

**Pull request merged**